flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/DefaultScheduler.java

```diff
@@ -160,7 +160,15 @@
 			failoverStrategy,
 			restartBackoffTimeStrategy);
 		this.schedulingStrategy = schedulingStrategyFactory.createInstance(this, getSchedulingTopology());
-		this.executionSlotAllocator = checkNotNull(executionSlotAllocatorFactory).createInstance(getPreferredLocationsRetriever());
+
+		final ExecutionSlotAllocationContext slotAllocationContext = new ExecutionSlotAllocationContext(
```

Comment:
Maybe this way it is easier  for testing.
I was more thinking about an interface `ExecutionSlotAllocationContext` implemented internally in `DefaultScheduler` by wrapping `preferredLocationsRetriever/jobGraph/EG/schedulingTopology`.

Reply:
done.